goal_src/engine/camera/cam-update.gc

```diff
@@ -5,3 +5,735 @@
+     )
+    (set! (-> v1-136 vector 3 quad) (the-as uint128 a3-12))
+    )
+   )
+  (let ((gp-3 (-> *math-camera* camera-temp))
+        (s4-1 (-> *math-camera* camera-rot))
+        (s5-2 (-> *math-camera* inv-camera-rot))
+        (s3-0 (-> *math-camera* trans))
+        )
+   (let ((s2-0 (new-stack-vector0)))
+    (set! (-> s2-0 x) (- (-> s3-0 x)))
+    (set! (-> s2-0 y) (- (-> s3-0 y)))
+    (set! (-> s2-0 z) (- (-> s3-0 z)))
+    (set! (-> s2-0 w) 1.0)
+    (vector-matrix*! s2-0 s2-0 s4-1)
+    (set! (-> s4-1 vector 3 quad) (-> s2-0 quad))
+    )
+   (matrix*! gp-3 s4-1 (-> *math-camera* perspective))
+   (set! (-> s5-2 vector 3 quad) (-> s3-0 quad))
+   )
+  (when (nonzero? (-> *math-camera* reset))
+   (let ((v1-148 (-> *math-camera* prev-camera-temp)))
+    (let* ((a3-15 (-> *math-camera* camera-temp))
+           (a0-54 (-> a3-15 vector 0 quad))
+           (a1-23 (-> a3-15 vector 1 quad))
+           (a2-19 (-> a3-15 vector 2 quad))
+           )
+     (set! a3-12 (-> a3-15 vector 3 quad))
+     (set! (-> v1-148 vector 0 quad) a0-54)
+     (set! (-> v1-148 vector 1 quad) a1-23)
+     (set! (-> v1-148 vector 2 quad) a2-19)
+     )
+    (set! (-> v1-148 vector 3 quad) (the-as uint128 a3-12))
+    )
+   (set! (-> *math-camera* reset) 0)
+   0
+   )
+  (let ((f0-45 (-> *math-camera* fog-min))
+        (f1-10 (-> *math-camera* fog-max))
+        )
+   (let ((f2-0 (-> *math-camera* d)))
+    (set! (-> *instance-tie-work* hmge-d x) f0-45)
+    (set! (-> *instance-tie-work* hmge-d y) f1-10)
+    (set! (-> *instance-tie-work* hmge-d z) (* 32.0 f2-0))
+    (set!
+     (-> *instance-tie-work* hmge-d w)
+     (* f2-0 (-> *math-camera* hmge-scale w))
+     )
+    (let ((v1-160 (-> *math-camera* hvdf-off quad)))
+     (set! (-> *instance-tie-work* hvdf-offset quad) v1-160)
+     )
+    (set! (-> *instance-shrub-work* hmge-d x) f0-45)
+    (set! (-> *instance-shrub-work* hmge-d y) f1-10)
+    (set! (-> *instance-shrub-work* hmge-d z) (* 3.0 f2-0))
+    (set!
+     (-> *instance-shrub-work* hmge-d w)
+     (* f2-0 (-> *math-camera* hmge-scale w))
+     )
+    (set!
+     (-> *instance-shrub-work* billboard-const x)
+     (/ (-> *math-camera* x-pix) (* f2-0 (-> *math-camera* x-ratio)))
+     )
+    (set!
+     (-> *instance-shrub-work* billboard-const y)
+     (/ (-> *math-camera* y-pix) (* f2-0 (-> *math-camera* y-ratio)))
+     )
+    )
+   (set! (-> *instance-shrub-work* billboard-const z) f0-45)
+   (set! (-> *instance-shrub-work* billboard-const w) f1-10)
+   )
+  (set!
+   (-> *instance-shrub-work* constants w)
+   (the-as float (-> *math-camera* vis-gifs 0 fog0))
+   )
+  (let ((a0-56 (-> *math-camera* hvdf-off quad)))
+   (set! (-> *instance-shrub-work* hvdf-offset quad) a0-56)
+   )
+  (update-view-planes *math-camera* (-> *math-camera* plane) 1.0)
+  (let ((t9-22 update-view-planes)
+        (a0-58 *math-camera*)
+        (a1-25 (-> *math-camera* guard-plane))
+        (a2-21 4.0)
+        )
+   (t9-22 a0-58 a1-25 a2-21)
```

Comment:
This is kinda suspicious, but I think will go away once `update-visible` takes 1 arg